pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -117,7 +117,7 @@ func NewController(
 	xdsUpdater xdsConfigUpdater,
 ) *Controller {
 	stop := make(chan struct{})
-	opts := krt.NewOptionsBuilder(stop, "ingress", options.KrtDebugger)
+	opts := krt.NewOptionsBuilder(stop, "ingress", options.KrtDebugger, nil)
```

Comment:
nit: rather than changing `NewOptionsBuilder` signature which affects every usage, the spot in ambeintindex could do something like

opts := NewOptionsBuider
opts = opts.With(metadata)?

Reply:
Problem is that the With functions return collection options not options builders. So that won't work even though it's cleaner 😕

Reply:
Made an exception to the pattern and just created a WithMetadata builder function